### ISSUE TITLE
Nested tensor: Support Discontinuous Buffer

### ISSUE THREAD

**Issue by YifanShenSZ** [pytorch/pytorch#80985]
### 🚀 The feature, motivation and pitch

Current implementation of nested tensor assume the buffer memory to be contiguous. However, some operations can break that, e.g.:
* reshape
* transpose
* slice

To deal with those cases, metadata strides and offsets are introduced in [#79831](https://github.com/pytorch/pytorch/pull/79831) and [#80981](https://github.com/pytorch/pytorch/pull/80981). 

Let's update existing nested tensor functionalities that assumed continuity to embrace the discontinuity and make use of the new metadata.

### Alternatives

For example, the way to access unbinded tensors from discontinuous buffer is `buffer.as_strided(size, stride, offset)`

**Comment by YifanShenSZ** [comment]
Here is a table tracking the update progress. Feel free to edit

| functionality | pr that updates it |
|--------------|------------------|
| unbind | |
| select | |
| to_padded_tensor | |
| dropout | |
| softmax | |
| bmm | |
